### DISCUSSION TITLE
Building TeaVM with WASI and Modifications to `org.teavm.backend.wasm.wasi.Wasi`

### DISCUSSION THREAD

**Discussion by mandesero** [konsoletyper/teavm#967] [Q&A]
I'm trying to build TeaVM with WASI support, specifically with modifications to the `org.teavm.backend.wasm.wasi.Wasi` class. I have tried using the commands `./gradlew publishToMavenLocal` and `./gradlew :tools:wasm-disassembly:build`, but I don't see any artifacts in `.m2/repository/org/teavm/teavm-wasm*` or anything similar.

Additionally, I would like to build a Java example that uses the modified `org.teavm.backend.wasm.wasi.Wasi` class. Could you provide guidance on how to achieve this?

**Comment by konsoletyper** [comment]
> `./gradlew :tools:wasm-disassembly:build`

Where did you take this?

> `.m2/repository/org/teavm/teavm-wasm*`

Where did you take this?

> Additionally, I would like to build a Java example that uses the modified org.teavm.backend.wasm.wasi.Wasi class.

First, start with easy way, I.e. with building js with unmodified TeaVM, then try with less direct options with wasm. Learn `samples` directory, some of samples definitely compile to wasi.

**Reply by mandesero** [reply]
Sorry, I didn't go into the topic deep enough. I fixed my problems with building, thanks.
